src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/ValueUtilsSmi.cs

```diff
@@ -2899,6 +2907,10 @@ int length
 /*EnSDR*/{ _ ,  _ ,  _ , _ , _ , _  , _ , _ , _  , _  , _  , _  , _  , _ , _ , _  , _ , _  , _  , _ , _ , _  , _  , _  , _ , _  , _ , _ , _ ,  _ , X,  _  , _  , _  , _ , },/*IEnurerable<SqlDataRecord>*/
 /*TmSpn*/{ _ ,  _ ,  _ , _ , _ , _  , _ , _ , _  , _  , _  , _  , _  , _ , _ , _  , _ , _  , _  , _ , _ , _  , _  , _  , _ , _  , _ , _ , _ ,  _ , _,  _  , X  , _  , _ , },/*TimeSpan*/
 /*DTOst*/{ _ ,  _ ,  _ , _ , _ , _  , _ , _ , _  , _  , _  , _  , _  , _ , _ , _  , _ , _  , _  , _ , _ , _  , _  , _  , _ , _  , _ , _ , _ ,  _ , _,  _  , _  , _  , X , },/*DateTimeOffset*/
+#if NET6_0_OR_GREATER
```

Comment:
I have NO idea if this is "correct" - copied from the row for DateTime/TimeSpan

Reply:
Looks ok to me clr DateOnly to the various sql Date types and var. TimeOnly only to Timespan.
That entire thing is copied from a grid in the docs somewhere about type conversions but i've no idea if it's been updated to include dateonly and timeonly.

Reply:
I have no idea either... 👀 I'll trust that the tests are validating that this is correct.

Reply:
It seems to be used to map the compatibility of SQL data types with extended CLR type codes. I found the following documentation the most relevant, but it's challenging to infer from there!
[SQL-CLR type mapping](https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/sql/linq/sql-clr-type-mapping)